lib/lte_link_control/lte_lc.c

```diff
@@ -1265,11 +1250,18 @@ int lte_lc_func_mode_get(enum lte_lc_func_mode *mode)
 int lte_lc_func_mode_set(enum lte_lc_func_mode mode)
 {
 	switch (mode) {
+	case LTE_LC_FUNC_MODE_ACTIVATE_LTE:
+	case LTE_LC_FUNC_MODE_NORMAL: {
+		int err = enable_notifications();
+
+		if (err) {
+			LOG_ERR("Failed to enabled notifications, error: %d", err);
+			return err;
```

Comment:
Would it be better to remove the `return` and try to send the +CFUN command also if `enable_notifications()` fails? Now we don't even try to set the functional mode if `enable_notifications()` call fails.

Reply:
Possibly, I was in doubt on that one. I landed on leaving the return there, because while still setting CFUN might succeed and connect to the network, the application would never be notified of the established connection since +CEREG=5 is not set. But I'm open to change it.

Reply:
Yes, I don't know which way would be better. Most likely if +CEREG fails, also +CFUN will fail. I'm not sure why the first command would fail, but the second command succeed. So maybe this doesn't matter. :)